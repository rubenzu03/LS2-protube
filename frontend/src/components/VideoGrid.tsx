import { useState, useEffect } from 'react';
import { getEnv } from '../utils/Env';

const VideoGrid = () => {
  const [someData, setSomeData] = useState([]);

  useEffect(() => {
    fetch(getEnv().API_BASE_URL + '/someEndpoint')
      .then((res) => {
        return res.json();
      })
      .then((data) => {
        setSomeData(data);
      });
  }, []);

  return (
    <ul className="row g-4">
      {someData?.map((entity) => (
        <li>{entity}</li>
      ))}
    </ul>
  );
};

export default VideoGrid;
