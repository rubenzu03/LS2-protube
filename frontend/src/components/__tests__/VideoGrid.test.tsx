import { act, render } from '@testing-library/react';
import '@testing-library/jest-dom';

import VideoGrid from '../VideoGrid';
import fetchMock from 'jest-fetch-mock';

describe('this component', () => {
  beforeEach(() => {
    fetchMock.resetMocks(); // Reset mocks before each test
  });

  it('renders correctly', async () => {
    fetchMock.mockResponseOnce(JSON.stringify(['video 1 test', 'video 2 test']));
    const asFragment = await act(async () => {
      return render(<VideoGrid />).asFragment;
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
