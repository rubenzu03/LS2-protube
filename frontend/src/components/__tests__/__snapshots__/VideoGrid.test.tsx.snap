// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`this component renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="row g-4"
  >
    <li>
      video 1 test
    </li>
    <li>
      video 2 test
    </li>
  </ul>
</DocumentFragment>
`;
